src/models/CustomAction.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * LINKURIOUS CONFIDENTIAL
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-08-19.
+ */
+
+import {IDataSourceParams} from './Model';
+
+// Create
+export interface ICreateCustomActionParams extends IDataSourceParams {
+    name: string;
+    urlTemplate: string;
+    description?: string;
+}
+export interface ICreateCustomActionResponse extends ICustomAction {}
+
+// Delete
+export interface IDeleteCustomActionParams extends IDataSourceParams {
+    id: number;
+}
+
+// Update
+export interface IUpdateCustomActionParams extends IDataSourceParams {
+    id: number;
+    name?: string;
+    urlTemplate?: string;
+    description?: string;
+}
+export interface IUpdateCustomActionResponse extends ICustomAction {}
+
+// Get
+export interface IGetCustomActionsParams extends IDataSourceParams {}
+export interface IGetCustomActionsResponse{
+    [index: number]: ICustomAction;
+}
+
+// Entities
+export interface ICustomAction {
+    id: number;
+    name: string;
+    urlTemplate: string;
+    description?: string;
```

Comment:
empty description is the same as an undefined description. I would make it mandatory for simplicity.